src/shared/modules/loader/services/loader-interceptor.service.ts

```diff
@@ -0,0 +1,31 @@
+import { Injectable, Injector } from '@angular/core';
+import { HttpEvent, HttpInterceptor, HttpHandler, HttpRequest, HttpResponse } from '@angular/common/http';
+import { Observable, pipe } from 'rxjs';
+import { tap } from 'rxjs/operators';
+import { LoaderService } from './loader.service';
+@Injectable({
+  providedIn: 'root'
+})
+export class LoaderInterceptorService implements HttpInterceptor {
```

Comment:
i think we should at this point be updating the state rather than having state in the service... so basically, the interceptor should be updating the state and loader should be observing that flag on that state... 
this approach would be good if we wouldn't have ngrx in place...

Reply:
Iterceptor service doesn't have any state in it